### PR TITLE
common, log: Extended message on death to include register values and  location of instruction pointer. 

### PR DESCRIPTION
**Pull request by aclamk**
Only for x86_64.           
Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

### CODE REVIEW

**Comment by aclamk** [comment]
Output now looks like this:
 ceph version 13.0.0-4829-g9a9a00c (9a9a00cb3de9b7cc2f1129b2d8f6f14e48dba321) mimic (dev)
 1: /home/adam/ceph-4/build/bin/ceph-osd (()+0x8c8266) [0x561aa03c1266]
 2: /lib/x86_64-linux-gnu/libpthread.so.0 (()+0x11630) [0x7f5fa392f630]
 3: /usr/lib/x86_64-linux-gnu/libjemalloc.so.1 (()+0xa3e4) [0x7f5fa4f9d3e4]
 4: /usr/lib/x86_64-linux-gnu/libjemalloc.so.1 (()+0xbdae) [0x7f5fa4f9edae]
 5: /usr/lib/x86_64-linux-gnu/libjemalloc.so.1 (()+0xbe87) [0x7f5fa4f9ee87]
 6: /usr/lib/x86_64-linux-gnu/libjemalloc.so.1 (()+0xbeda) [0x7f5fa4f9eeda]
 7: /usr/lib/x86_64-linux-gnu/libjemalloc.so.1 (()+0xd937) [0x7f5fa4fa0937]
 8: /usr/lib/x86_64-linux-gnu/libjemalloc.so.1 (()+0x59a1) [0x7f5fa4f989a1]
 9: /home/adam/ceph-4/build/lib/libceph-common.so.0 (ceph::buffer::create_aligned_in_mempool(unsigned int, unsigned int, int)+0xe8) [0x7f5fa62e2bd8]
 a: /home/adam/ceph-4/build/bin/ceph-osd (KernelDevice::aio_read(unsigned long, unsigned long, ceph::buffer::list*, IOContext*)+0x4d0) [0x561aa03afa70]
 b: /home/adam/ceph-4/build/bin/ceph-osd (()+0x75b690) [0x561aa0254690]
 c: /home/adam/ceph-4/build/bin/ceph-osd (bluestore_blob_t::map(unsigned long, unsigned long, std::function<int (unsigned long, unsigned long)>) const+0xea) [0x561aa02d073a]
 d: /home/adam/ceph-4/build/bin/ceph-osd (BlueStore::_do_read(BlueStore::Collection*, boost::intrusive_ptr<BlueStore::Onode>, unsigned long, unsigned long, ceph::buffer::list&, unsigned int)+0x19bc) [0x561aa02b842c]
 e: /home/adam/ceph-4/build/bin/ceph-osd (BlueStore::_fsck(bool, bool)+0x4263) [0x561aa02c8f83]
 f: /home/adam/ceph-4/build/bin/ceph-osd (BlueStore::_mount(bool, bool)+0x126) [0x561aa02cd226]
 10: /home/adam/ceph-4/build/bin/ceph-osd (OSD::init()+0x387) [0x561a9fe78d87]
 11: /home/adam/ceph-4/build/bin/ceph-osd (main()+0x38ae) [0x561a9fd6ab7e]
 12: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main()+0xf1) [0x7f5fa28b13f1]
 13: /home/adam/ceph-4/build/bin/ceph-osd (_start()+0x2a) [0x561a9fe34d6a]
rax 0x7fffd0ca4e40
rbx 0x0 0
rcx 0x0 0
rdx 0x7f5f14801608
rsi 0x7f5f0f0025f8
rdi 0x7f5fa20001c0
rsp 0x7fffd0ca4e00
rbp 0x189 393
r8  0x310000 3211264
r9  0x7fffd0ca4e00
r10 0x216000 2187264
r11 0x7f5fa20001c8
r12 0x8 8
r13 0x10000 65536
r14 0x7f5fa51c9370
r15 0x0 0
rip 0x7f5fa4f9d3e4
IP is at /usr/lib/x86_64-linux-gnu/libjemalloc.so.1:():a3e4

File: src/common/BackTrace.cc [hunk 1]
```diff
@@ -10,63 +10,63 @@
...
-      free(foo);
-    } else {
-      // didn't find the mangled name, just print the whole line
-      out << " " << (i-skip+1) << ": " << strings[i] << std::endl;
-    }
-    free(function);
+    std::string_view object_elf;
+    std::string function;
+    uint64_t offset;
+    std::tie(object_elf, function, offset) = split_backtrace_line(strings[i]);
```
Comment (tchaikov) [inline src/common/BackTrace.cc]: no need to use `std::tie()` anymore, as we can use structured binding in C++17, like
```c++
auto [object_elf, function, offset] = split_backtrace_line(strings[i]);
```

Comment (aclamk) [reply src/common/BackTrace.cc]: Thx. I wasn't aware of.

**Comment by neha-ojha** [comment]
@aclamk let's rebase this one and ensure it passes `make check`

**Comment by tchaikov** [comment]
retest this please.

**Comment by aclamk** [comment]
retest this please.

**Comment by tchaikov** [comment]
retest this please

**Pull request closed without merging**